### PR TITLE
add yardoc under Ruby documentation

### PR DESCRIPTION
**Pull request by mejackreed**
Starts to address #11 

Most likely there will be different documentation standards for different languages. Specifically in mind here is JavaScript

### CODE REVIEW

File: best-practices/documentation/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Documentation
+
+## Ruby
+Ruby projects should use [Yard](http://yardoc.org/) for documentation purposes.
+
+One benefit is YARD runs a live documentation server for the community, hosting docs for all RubyGems and Github projects. The project is at [RubyDoc.info](http://www.rubydoc.info/) (formerly rdoc.info)
+
+A [useful Yardoc cheatsheet](https://gist.github.com/chetan/1827484)
```
Comment (cbeer) [inline best-practices/documentation/README.md]: Should we also link to http://www.rubydoc.info/gems/yard/file/docs/GettingStarted.md?

Comment (cbeer) [inline best-practices/documentation/README.md]: Do you know something similar to http://guides.rubyonrails.org/api_documentation_guidelines.html, but for YARD? I like it's focus on the content of the comment, and maybe it just needs to be adapted for YARD.

Comment (mejackreed) [inline best-practices/documentation/README.md]: @cbeer linked the getting started doc. No idea about anything in regards to content of comment. Maybe @atz, @ndushay,  or @jmartin-sul know of something? They seem to have some good ideas about commenting content.

**Comment by cbeer** [comment]
:+1:

**Comment by ndushay** [comment]
:+1:   eventually may want more here, but this is a great start.

**Comment by darrenleeweber** [comment]
:+1: on using yard, but wondering whether the generic `README.md` with language headings might be split into separate documents for each language, i.e. `documentation/ruby.md` and `documentation/javascript.md` etc. with the `README.md` linking to each.

**Comment by mejackreed** [comment]
Updated to deal with merge conflict... no content changed.

**Comment by cbeer** [comment]
:+1: YARD is pretty great, and I notice this PR says nothing about how it is used or metrics to quibble over.

I disagree about splitting into different documents. It seems premature, and we can always reorganize it if/when it becomes unwieldy.

**Pull request merged**